test/e2e/framework/node/resource.go

```diff
@@ -559,6 +562,14 @@ func CreatePodsPerNodeForSimpleApp(c clientset.Interface, namespace, appName str
 	return podLabels
 }
 
+// AddOrUpdateTaintOnNode adds the given taint to the given node or updates taint.
+func AddOrUpdateTaintOnNode(c clientset.Interface, nodeName string, taint v1.Taint) {
+	err := controller.AddOrUpdateTaintOnNode(c, nodeName, &taint)
+
+	// TODO use wrapper methods in expect.go after removing core e2e dependency on node
```

Comment:
> TODO use wrapper methods in expect.go after removing core e2e dependency on node

What do you mean by "core e2e dependency on node"?

Reply:
that means framework core package  can't  dependency sub-package /test/e2e/framework/node , the core package should not depend on the subpackage as much as possible ,the topic is from  https://github.com/kubernetes/kubernetes/issues/81245 .

Reply:
Move E2E Framework into Staging has higher priority.

Reply:
after moving this function , i will remove the direct dependency for internal k8s.io/kubernetes
removing the direct dependency for internal k8s.io/kubernetes

Reply:
1. for ```ExpectNoError(controller.AddOrUpdateTaintOnNode(c, nodeName, &taint))```  , I can't directly use framework.ExpectNoError due to ``` import cycle not allowed ```. so I leave some comment, like ``` // TODO use wrapper methods in expect.go after removing core e2e dependency on node ```.
2. “core e2e dependency” means I understand that subpackages can dependency core packages but the core package should not depend on the subpackage. there is a open issue https://github.com/kubernetes/kubernetes/issues/81245 to record this problem .see https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L55, https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L64
3. https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L67 , it will result in  ``` import cycle not allowed ```  if we use framework.Expect*() in subpackage test/e2e/framework/node.
4. My ultimate goal is to remove the reference to the controller, see https://github.com/kubernetes/kubernetes/pull/90190/files#diff-5ecf81cdd161eaf9b17afda938ddcbe5R567, so I will first move this function into a sub-package test/e2e/framework/node, and then complete my final goal to make the pr function as independent as possible.

Reply:
@tanjunchen has good answer now.
@alejandrox1 Could you check this PR again?

Reply:
/cc @alejandrox1

Reply:
I will update the conflict after we get the answer.

Reply:
Excuse the delay. Coming around to, and I apologize for the vagueness. I understand the issue here but my comment was about properly communicating it in the comment
```
TODO use wrapper methods in expect.go after removing core e2e dependency on node
```

that comment does not properly communicate what you have been discussing here @tanjunchen .
If we are going to leave TODOs in the code base we should also assume that we will forget about them and someone else will eventually find them and try to resolve them.
The comment you have here does fully communicate the work to be done.

A better comment could be something like
```
TODO use wrapper methods in expect.go after removing the dependency on this 
package from the core e2e framework.
``` 

My comment here is just for us to standardize on the language we used to talk about things.

Reply:
get it,Thanks for your comment,i will update it.

Reply:
@oomichi @alejandrox1 
done. pls review this again! Thanks